ignition-server/controllers/tokensecret_controller.go

```diff
@@ -249,11 +254,13 @@ func (r *TokenSecretReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	compressedConfig := tokenSecret.Data[TokenSecretConfigKey]
 	config, err := util.DecodeAndDecompress(compressedConfig)
 	if err != nil {
-		patch := tokenSecret.DeepCopy()
-		patch.Data[TokenSecretReasonKey] = []byte(InvalidConfigReason)
-		patch.Data[TokenSecretMessageKey] = []byte(fmt.Sprintf("Failed to decode and decompress config: %s", err))
-		if err := r.Client.Patch(ctx, patch, client.MergeFrom(tokenSecret)); err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to patch tokenSecret with payload content: %w", err)
+		if !hasSameReasonAndMessage(tokenSecret, InvalidConfigReason, fmt.Sprintf("Failed to decode and decompress config: %s", err)) {
```

Comment:
nit: can we instead return early if hasSameReasonAndMessage, and otherwise run the bulk logic